### PR TITLE
GEOMESA-2197 Adding geomesa process visitor handling for HBase

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by jahhulbert-ccri** [comment]
any testing that needs to be done?

**Comment by elahrvivaz** [comment]
i verified that the implementations are in place, but yeah, i should add some unit tests.

File: geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseFeatureCollection.scala [hunk 1]
```diff
@@ -27,12 +24,8 @@ class HBaseFeatureCollection(source: GeoMesaFeatureSource, query: Query)
 
   override def accepts(visitor: FeatureVisitor, progress: ProgressListener): Unit =
     visitor match {
-      case v: QueryVisitor     => v.execute(source, query)
-      case v: AttributeVisitor => v.execute(source, query)
-      case v: ArrowVisitor     => v.execute(source, query)
-      case v: BinVisitor       => v.execute(source, query)
-      case v: StatsVisitor     => v.execute(source, query)
-      case v: MinMaxVisitor    => v.execute(source, query)
```
Comment (jnh5y) [inline geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseFeatureCollection.scala]: Remind me; do all these classes implement GeoMesaProcessVisitor?

Comment (elahrvivaz) [reply geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseFeatureCollection.scala]: they do, yeah

**Comment by jahhulbert-ccri** [comment]
i'll merge after unit tests if you want to add

**Pull request merged**